### DISCUSSION TITLE
Existe uma forma de um component filho herdar todos as properties do pai de um componente Blazor?

### DISCUSSION THREAD

**Discussion by paulinhps** [balta-io/forum#427] [Tira Dúvidas] [answered]
Olá, em outras linguagens e frameworks que usam o conceito de componentes é comum criar componentes que possuem outros componentes filhos e ter a opção de "injetar" todos os parâmetros do pai. Porém eu não consegui encontrar em um componente Blazor uma forma de fazer isso.

```cshtml
@*<NavLink class="nav-link" href="" Match="NavLinkMatch.All"> - Exemplo da configuração do componente*@
<NavLink @*class="nav-link" href="" Match="NavLinkMatch.All" *@> @*Gostaira de Injetar todas as propriedades que são desse (this) componente*@
    <span class="oi oi-home" aria-hidden="true"> </span> @Title
</NavLink>
@code {

    [Parameter]
    public RenderFragment? Title { get; set; }
}

```

Então eu poderia fazer a chamada desse componente assim:

```cshtml
<NavItem class="nav-link" href="" Match="NavLinkMatch.All">
       <Title> Title</Title>
 </NavItem>
```

> [!NOTE]  
> O exemplo acima é bem corriqueiro e não usaria, é apenas para expressar a minha dúvida sobre passagem de parâmetros entre componentes Blazor

Muito Obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @paulinhps, como vai?

Pode sim, no Blazor isto é feito através do `ChildContent`, da mesma forma que React, Angular, Vue e etc....

```csharp
@* MeuComponente.razor *@
<div>
    @ChildContent
</div>

@code {
    [Parameter]
    public RenderFragment? ChildContent { get; set; }
}
```

```csharp
@page "/sua-pagina"

<MeuComponente>
    Conteúdo que vai aparecer dentro do componente
</MeuComponente>
```

**Reply by paulinhps** [reply]
Ok essa parte eu entendi! Mas ainda não conseguir fazer a passagem "full" dos parâmetros como é feito no React por exemplo.

```cshtml

@* MyPage.razor *@
<MyComponent title="html title" id="htmlId" @onmouseover="OnMouseHoverHandler"></MyComponent>
```



```cshtml

@* MyComponent.razor *@
<AnythingComponent @* inject here all parents (title, id, @onmouseover) *@ @onclick="MyClickHandler">
    @* code... *@
</AnythingComponent>
```

Ou realmente eu tenho que deixar qualquer parâmetro explicito?

**Reply by andrebaltieri** [reply]
Você pode criar um objeto em comum entre estes dois componentes e criar uma propriedade em cada um deles com este objeto.

```csharp
public class MyObject { ... }

// No componente 1
[Parameter]
public MyObject Obj {get; set;}

// No componente 2
[Parameter]
public MyObject Obj {get; set;}
```

Ai você pode reusar o objeto no Razor:
```csharp
<MyComponent1 Obj="myObj" />
<MyComponent2 Obj="myObj" />
```

Como são classes, ou tipos de referência, você terá apenas a passagem de referência entre os componentes e só um objeto na memória.
